presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoLdapCliTests.java

```diff
@@ -0,0 +1,273 @@
+import com.teradata.tempto.Requirement;
+import com.teradata.tempto.RequirementsProvider;
+import com.teradata.tempto.Requires;
+import com.teradata.tempto.configuration.Configuration;
+import com.teradata.tempto.fulfillment.ldap.LdapObjectRequirement;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+
+import static com.facebook.presto.tests.ImmutableLdapObjectDefinitions.AMERICA_ORG;
+import static com.facebook.presto.tests.ImmutableLdapObjectDefinitions.ASIA_ORG;
+import static com.facebook.presto.tests.ImmutableLdapObjectDefinitions.CHILD_GROUP;
+import static com.facebook.presto.tests.ImmutableLdapObjectDefinitions.CHILD_GROUP_USER;
+import static com.facebook.presto.tests.ImmutableLdapObjectDefinitions.DEFAULT_GROUP;
+import static com.facebook.presto.tests.ImmutableLdapObjectDefinitions.DEFAULT_GROUP_USER;
+import static com.facebook.presto.tests.ImmutableLdapObjectDefinitions.ORPHAN_USER;
+import static com.facebook.presto.tests.ImmutableLdapObjectDefinitions.PARENT_GROUP;
+import static com.facebook.presto.tests.ImmutableLdapObjectDefinitions.PARENT_GROUP_USER;
+import static com.facebook.presto.tests.TestGroups.ACTIVE_DIRECTORY;
+import static com.facebook.presto.tests.TestGroups.LDAP;
+import static com.facebook.presto.tests.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static com.teradata.tempto.process.CliProcess.trimLines;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.Objects.requireNonNull;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.testng.Assert.assertTrue;
+
+public class PrestoLdapCliTests
+    extends PrestoCliTests
+    implements RequirementsProvider
+{
+    @Inject(optional = true)
+    @Named("databases.presto.cli_ldap_truststore_path")
+    private String ldapTruststorePath;
+
+    @Inject(optional = true)
+    @Named("databases.presto.cli_ldap_truststore_password")
+    private String ldapTruststorePassword;
+
+    @Inject(optional = true)
+    @Named("databases.presto.cli_ldap_user_name")
+    private String ldapUserName;
+
+    @Inject(optional = true)
+    @Named("databases.presto.cli_ldap_server_address")
+    private String ldapServerAddress;
+
+    @Inject(optional = true)
+    @Named("databases.presto.cli_ldap_user_password")
+    private String ldapUserPassword;
+
+    public PrestoLdapCliTests()
+            throws IOException
+    {}
+
+    @Override
+    public Requirement getRequirements(Configuration configuration)
+    {
+        return new LdapObjectRequirement(
+                Arrays.asList(
+                        AMERICA_ORG, ASIA_ORG,
+                        DEFAULT_GROUP, PARENT_GROUP, CHILD_GROUP,
+                        DEFAULT_GROUP_USER, PARENT_GROUP_USER, CHILD_GROUP_USER, ORPHAN_USER
+                ));
+    }
+
+    @AfterTestWithContext
+    public void stopPresto()
+            throws InterruptedException
```

Comment:
I tried removing this method from the subclass. The annotated method from the superclass isn't being called. So will keep it as-is.